2-mandatory/4-people-i-know.js

```diff
@@ -377,14 +377,36 @@ In the above object you can see my friends and the colleagues of my friends.
 First, I want you to find all of my friends who are 35 or older.
 */
 
-let thirtyFiveOrOlder = [];
+let thirtyFiveOrOlder = friends.filter(function (friend) {
+  if (friend.age >= 35) {
+    return friend.name.first;
+  }
+});
+
+// ### SOLUTION - CYF
+// let thirtyFiveOrOlder = friends.filter((friend) => friend.age >= 35);
 
 /*
 3) Find the email address
 Next, I want you to find all of my friends who work for "POWERNET" and then store their emails in the array below
 */
 
 let powerNetEmails = [];
+friends
+  .filter((eml) => eml.email)
+  .reverse()
+  .filter(function (friendCompany) {
+    if (friendCompany.company === "POWERNET") {
+      return powerNetEmails.push(friendCompany.email);
+    }
+  });
```

Comment:
1. The first filter at line 396 seems not doing much as all person objects has an email attribute
2. reverse() has no use here. Test might fail if it test both the appearance of correct elements and also the order of elements.
3. Variable naming. The object we passed to filter function is a person **object** which is better than naming it **friendCompany**